### PR TITLE
Build standard libraries with exceptions/RTTI disabled.

### PR DESCRIPTION
**Pull request by paulhuggett**
Communicating the --fno-exceptions and --fno-rtti switches (in addition to the existing --sysroot) to cmake via the Dockerfile has entailed to switching to using BuildKit's updated Dockerfile implementation (docker/dockerfile:1.3). This requires [Docker 18.09](https://docs.docker.com/develop/develop-images/build_enhancements/) or later.

Remove unused variables and steps obsoleted now that the compiler is “repo by default”.

### CODE REVIEW

File: llvm/utils/repo/docker/llvm-prepo/Dockerfile [hunk 1]
```diff
@@ -128,83 +123,105 @@ RUN mkdir -p $musl_install_dir/include/linux
...
+RUN repo-create-ticket                                         \
+  --output=${install_dir}/lib/linux/clang_rt.crtbegin-x86_64.o \
+  --repo=${stdlib_repo}                                        \
+  d3540c32449fa62681dc5c22fbc2a353
+RUN repo-create-ticket                                       \
+  --output=${install_dir}/lib/linux/clang_rt.crtend-x86_64.o \
+  --repo=${stdlib_repo}                                      \
+  d3540c32449fa62681dc5c22fbc2a353
+
+RUN ar d ${install_dir}/lib/linux/libclang_rt.builtins-x86_64.a
```
Comment (MaggieYingYi) [inline llvm/utils/repo/docker/llvm-prepo/Dockerfile:141]: It looks like the the file `${install_dir}/lib/linux/clang_rt.o` is removed from the command line. Is it correct?

Comment (paulhuggett) [reply llvm/utils/repo/docker/llvm-prepo/Dockerfile:141]: Yes. This command is trying to create an empty archive file and using the ‘d’ command to do it. Deleting clang_rt.o from an archive that doesn’t exist doesn’t make much sense.

Comment (paulhuggett) [reply llvm/utils/repo/docker/llvm-prepo/Dockerfile:141]: Actually, I missed that we are doing something similar on two later lines. I‘ll correct them as well.

**Review by MaggieYingYi — APPROVED** [review]
LGTM!

**Pull request merged**